lms/djangoapps/ccx/utils.py

```diff
@@ -4,283 +4,22 @@
-        ),
-        'allowed_unenroll': (
-            'ccx/unenroll_email_subject.txt',
-            'ccx/unenroll_email_allowedmessage.txt'
-        ),
-        'enrolled_unenroll': (
-            'ccx/unenroll_email_subject.txt',
-            'ccx/unenroll_email_enrolledmessage.txt'
-        ),
-    }
-
-    subject_template, message_template = email_template_dict.get(
-        message_type, (None, None)
-    )
-    if subject_template is not None and message_template is not None:
-        subject = render_to_string(subject_template, param_dict)
-        message = render_to_string(message_template, param_dict)
-
-    if subject and message:
-        message = message.strip()
-
-        subject = ''.join(subject.splitlines())
-        from_address = microsite.get_value(
-            'email_from_address',
-            settings.DEFAULT_FROM_EMAIL
-        )
-
-        send_mail(
-            subject,
-            message,
-            from_address,
-            [student],
-            fail_silently=False
-        )
-
-
-def get_ccx_membership_triplets(user, course_org_filter, org_filter_out_set):
-    """
-    Get the relevant set of (CustomCourseForEdX, CcxMembership, CourseOverview)
-    triplets to be displayed on a student's dashboard.
-    """
-    error_message_format = "User {0} enrolled in {2} course {1}"
-
-    # only active memberships for now
-    for membership in CcxMembership.memberships_for_user(user):
-        ccx = membership.ccx
-
-        try:
-            course_overview = CourseOverview.get_from_id(ccx.course_id)
-        except CourseOverview.DoesNotExist:
-            log.error(error_message_format.format(  # pylint: disable=logging-format-interpolation
-                user.username, ccx.course_id, "non-existent"
-            ))
-            continue
-        except IOError:
-            log.error(error_message_format.format(  # pylint: disable=logging-format-interpolation
-                user.username, ccx.course_id, "broken"
-            ))
-            continue
-
-        # if we are in a Microsite, then filter out anything that is not
-        # attributed (by ORG) to that Microsite
-        if course_org_filter and course_org_filter != course_overview.location.org:
-            continue
-        # Conversely, if we are not in a Microsite, then let's filter out any enrollments
-        # with courses attributed (by ORG) to Microsites
-        elif course_overview.location.org in org_filter_out_set:
-            continue
-
-        # If, somehow, we've got a ccx that has been created for a
-        # course with a deprecated ID, we must filter it out. Emit a
-        # warning to the log so we can clean up.
-        if course_overview.location.deprecated:
-            log.warning(
-                "CCX %s exists for course %s with deprecated id",
-                ccx,
-                ccx.course_id
-            )
-            continue
-
-        yield (ccx, membership, course_overview)
+    return ccx and ccx[0]
```

Comment:
It might be clearer to just explictly return `None` after your call to `warning`, rather than using the inline `and` here.